### DISCUSSION TITLE
First governance solution

### DISCUSSION THREAD

**Discussion by fermentfan** [veramolabs/did-eth#12] [Ideas]
I am thinking whether we should already get some DAO governance solution like a Gnosis Safe which simply is a multi sig wallet that we could leverage for simple fund collection and deploying to mainnet (as it's should be a secured deployment from the beginning?).
Also I'd like to transfer over the 'did-dao.eth' if we're interested in leveraging that name.

**Comment by Hmac512** [comment]
We should probably take our time with determining who will be a member of the safe in the beginning. I’m pretty sure this topic is likely to spark heated debates.

Unless we add a lot of people with a threshold requirement.

I think we should think about governance first. How to upgrade the contract, how to make proposals, how to vote on proposals etc.

I think a governance token might be too much, but the nice thing about a governance token is the people using the contract have a say in the governance.

Who wants to take on the role of researching governance?

**Comment by italobb** [comment]
I also suggest treating "DAO governance" entirely separately from "DID method governance".

If we arrive at a truly decentralized solution for the DID method, no matter who deployed a smart contract used in it, it will be trusted and accepted by the community after being audited.

Having an upgradeable solution that can be upgraded by a select number of people is neither secure nor decentralized. It's not a multi-sig what solves this issue.

Making a parallel with the Ethereum Foundation. They suggest the new protocol updates, but the network decides whether or not to adopt the new node implementations. We cannot simply have a multi-sig deciding whether a smart contract should be updated and change the behavior of the DID method.

**Reply by Hmac512** [reply]
One thing I have in mind is to set up the diamonds to let anyone add a resolver. It’s just mapping the strings like “did:eth” or “did:pkh” are the matter of governance (https://github.com/veramolabs/did-eth/issues/8)


I think we should build the version of the contract that is centrally managed first though. As an intermediate step to the truly trust-less version.


In summary though, the contract is updatable but people still have the choice to keep using the previous version of the contract.

There is still some governance involved, but it’s not as rigid as a multisig that controls everything.
